File: 이민호/큰수만들기.js
function solution(number, k) {
    var answer = '';
    const stack = [];
    for(let i = 0; i < number.length; i++)
    {
        const n = number[i];
        while(k > 0 && stack.length > 0 && stack[stack.length - 1] < n)
        {
            stack.pop();
            k--;
        }
        stack.push(n)
    }
    stack.splice(stack.length - k, k);
    answer = stack.join("");
    return answer;
}
const r = solution("1924",2);
console.log(r);

File: 김민지/programmers/큰 수 만들기.py
# 2시간 43분

def solution(number, k):
  sortNum = ''.join(sorted(number, reverse=True))
  if sortNum == number:
    return number[:len(number)-k]
  
  result = []
  for num in number:
    while k>0 and result and result[-1]<num:
      result.pop()
      k -= 1

    result.append(num)
    
  return ''.join(result)

print(solution("9999999999", 5))